### ISSUE TITLE
Accessibility - Reduce Motion

### ISSUE THREAD

**Issue by dw2kim** [flutter/flutter#65874]

**Comment by jonahwilliams** [comment]
Reduce motion and disable animations are not equivalent.

**Comment by dw2kim** [comment]
@jonahwilliams Thanks for clarifying! Is there any way to detect the Reduce Motion from the app? Or, is there a reason why `reduceMotion` is not exposed to MediaQueryData?

**Comment by jonahwilliams** [comment]
reduce motion is exposed through the window's AccessibilityFeatures. We decided to stop putting _everything_ on media query after a point.

**Comment by dw2kim** [comment]
@jonahwilliams I see. I think the `reduceMotion` is only flag that is missing in the media query for the Accessibility.

 Would it be okay to open PR just to expose this `reduceMotion` flag? 
Or, is there a better way to access this flag from the App?

**Comment by jonahwilliams** [comment]
I think @HansMuller had some idea on this. I believe we should make a new inherited widget that only exposes the accessibility features, and prefer those over the existing ones on media query.

**Comment by HansMuller** [comment]
This a good suggestion for improving Flutter's a11y support. Factoring reduceMotion into the widget implementations would take a considerable effort.

CC @chunhtai

**Comment by kennethj** [comment]
Any updates on this?

**Comment by chunhtai** [comment]
There is no progress done on this issue. Since there is already a way to get the reduce motion setting, this is a low priority issue. We welcome any contribution though.

**Comment by lukepighetti** [comment]
@chunhtai how does one get the reduce motion setting? I assumed `disableAnimations` was Reduce Motion in iOS accessibility settings but it doesn't appear to be. I see no way to obtain it.

**Comment by lukepighetti** [comment]
> Reduce motion and disable animations are not equivalent.


I think it's reasonable to wire up Reduce Motion to `disableAnimations`, considering the current documentation 
 
> The platform is requesting that animations be disabled **_or simplified_**.

**Comment by Albert221** [comment]
@chunhtai would you see `reduceMotion` as an additional `MediaQueryData` field, or as @jonahwilliams suggested, we should create a separate inherited widget that is responsible just for these accessibility settings? I'm potentially willing to work on this :)

**Comment by chunhtai** [comment]
We can add to [`AccessibilityFeatures`](https://api.flutter.dev/flutter/semantics/SemanticsBinding/accessibilityFeatures.html)

**Comment by kerberjg** [comment]
`AccessibilityFeatures.reduceMotion` is now accessible as `bool` via 

```dart
View.of(context).platformDispatcher.accessibilityFeatures.reduceMotion
```

Can this be considered solved?

**Comment by tech-andgar** [comment]
> `AccessibilityFeatures.reduceMotion` is now accessible as `bool` via
> 
> ```dart
> View.of(context).platformDispatcher.accessibilityFeatures.reduceMotion
> ```
> 
> Can this be considered solved?

Checked, but only working in iOS.

**Comment by chrisbobbe** [comment]
I propose closing this, since this setting is now exposed via [`AccessibilityFeatures.reduceMotion`](https://api.flutter.dev/flutter/dart-ui/AccessibilityFeatures/reduceMotion.html).

> Checked and only working in iOS. But no working other platforms (Web, Desktop macOS)

It's not expected to work on other platforms, and the dartdoc is already clear about that:

```dart
  /// The platform is requesting that certain animations be simplified and
  /// parallax effects removed.
  ///
  /// Only supported on iOS.
  bool get reduceMotion => _kReduceMotionIndex & _index != 0;
```

This is an iOS setting. Here are the Apple user and developer docs about it:

- https://support.apple.com/guide/iphone/customize-onscreen-motion-iph0b691d3ed/ios
- https://developer.apple.com/documentation/swiftui/environmentvalues/accessibilityreducemotion

If other platforms have similar settings and there's interest in exposing those, I think we can track them in separate issues.

**Comment by cbracken** [comment]
For what it's worth, I wired this through to MediaQuery in https://github.com/flutter/flutter/pull/190287
